kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/core/v1/ServiceOperationsImpl.java

```diff
@@ -58,30 +60,21 @@ public ServiceOperationsImpl newInstance(OperationContext context) {
 
   @Override
   public Service replace(Service item) {
-      try {
-        Service old = fromServer().get();
-        return super.replace(new ServiceBuilder(item)
-          .editSpec()
-          .withClusterIP(old.getSpec().getClusterIP())
-          .endSpec()
-          .build());
-      } catch (Exception e) {
-        throw KubernetesClientException.launderThrowable(forOperationType("replace"), e);
-      }
+    return super.replace(patchClusterIpIntoServiceAndReplace(item));
   }
 
   @Override
   public Service patch(Service item) {
-      try {
-        Service old = getMandatory();
-        return super.patch(new ServiceBuilder(item)
-          .editSpec()
-          .withClusterIP(old.getSpec().getClusterIP())
-          .endSpec()
-          .build());
-      } catch (Exception e) {
-        throw KubernetesClientException.launderThrowable(forOperationType("patch"), e);
-      }
+    try {
```

Comment:
Shouldn't the `patch` also work with `ExternalName`?

Reply:
I think you're right. Could you please create an issue for this?